R/plotColTile.R

```diff
@@ -0,0 +1,206 @@
+#' Plot factor data as tiles
+#' 
+#' Relative relations of two grouping can be visualized by plotting tiles with
+#' relative sizes. \code{plotColTile} and \code{plotRowTile} can be used for 
+#' this.
+#' 
+#' @param object a
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#' 
+#' @param x String specifying the column-level metadata field to show on the x-axis.
+#'   Alternatively, an \link{AsIs} vector or data.frame, see 
+#'   \code{?\link{retrieveFeatureInfo}} or \code{?\link{retrieveCellInfo}}. Must
+#'   result in a returned \code{character} or \code{factor} vector.
+#'   
+#' @param y String specifying the column-level metadata to show on the y-axis.
+#'   Alternatively, an \link{AsIs} vector or data.frame, see 
+#'   \code{?\link{retrieveFeatureInfo}} or \code{?\link{retrieveCellInfo}}. Must
+#'   result in a returned \code{character} or \code{factor} vector.
+#'   
+#' @param factor_x_by 
+#'  
+#' @param factor_y_by 
+#' 
+#' @return 
+#' A \code{ggplot2} object or \code{plotly} object, if more than one 
+#' \code{prevalences} was defined.
+#' 
+#' @name plotColTile
+#' 
+#' @examples
+#' data(GlobalPatterns)
+#' se <- GlobalPatterns
+#' plotColTile(se,"SampleType","Primer")
+NULL
+
+#' @rdname plotColTile
+#' @export
+setGeneric("plotColTile", signature = c("object"),
+           function(object, x, y, ...) standardGeneric("plotColTile"))
+#' @rdname plotColTile
+#' @export
+setGeneric("plotRowTile", signature = c("object"),
+           function(object, x, y, ...) standardGeneric("plotRowTile"))
+
+
+#' @rdname plotColTile
+#' @export
+setMethod("plotColTile", signature = c("SummarizedExperiment"),
+    function(object, x, y, factor_x_by = NULL, factor_y_by = NULL, ...){
+        .plot_tile_data(object, type = "column", x, y, 
+                        factor_x_by = factor_x_by, factor_y_by = factor_y_by,
+                        ...)
+    }
+)
+
+#' @rdname plotColTile
+#' @export
+setMethod("plotRowTile", signature = c("SummarizedExperiment"),
+    function(object, x, y, factor_x_by = NULL, factor_y_by = NULL, ...){
```

Comment:
The `plotRowTile` will be tricky because of the high number of variables. 
```
plotRowTile(tse, x="Phylum", y="Family")

```
It is an important function to check which taxa within a given rank are dominant and will be useful. One option I think is to maybe structure it like giving top 3-5 within each Phylum to be plotted with an `other` option? It maybe difficult to achieve this but do you think it is possible?

Reply:
I think this is the user's job. We cannot prevent `garbage in, garbage out`.